refinery/tool_manager/tests.py

```diff
@@ -2979,6 +2979,38 @@ def test_get_with_invalid_tool_type_request_param(self):
         self._make_tools_get_request(tool_type="coffee")
         self.assertEqual(self.get_response.data, [])
 
+    def _test_launch_vis_container(self, user_has_permission=True):
+        self.create_tool(ToolDefinition.VISUALIZATION)
+        self.assertFalse(self.tool.is_running())
+
+        if user_has_permission:
+            assign_perm('core.read_dataset', self.user, self.tool.dataset)
+
+        # Need to set_password() to be able to login. Otherwise
+        # usr.password in the hash representation which is not what the
+        # login() expects
+        temp_password = "password"
+        self.user.set_password(temp_password)
+        self.user.save()
+        self.client.login(username=self.user.username,
+                          password=temp_password)
+
+        with mock.patch.object(VisualizationTool, "launch") as launch_mock:
+            get_response = self.client.get(
+                "{}/".format(self.tool.get_relative_container_url())
```

Comment:
Is the trailing slash required?

Reply:
It is unfortunately, Normal django routing middleware tacks this on automatically. The `format()` is probably overkill for this though